source/slang/parameter-binding.cpp

```diff
@@ -977,7 +977,7 @@ getTypeLayoutForGlobalShaderParameter_GLSL(
 
     // We want to check for a constant-buffer type with a `push_constant` layout
     // qualifier before we move on to anything else.
-    if( varDecl->HasModifier<GLSLPushConstantLayoutModifier>() && type->As<ConstantBufferType>() )
+    if( varDecl->HasModifier<PushConstantAttribute>() && type->As<ConstantBufferType>() )
```

Comment:
As written, this will catch the `[[vk::...]]` attribute even when compiling for a non-VK target. This isn't a huge problem because D3D/DXIL doesn't currently differentiate push constant buffers from ordinary constant buffers, but I imagine it could cause a problem if we ever had distinct `[[vk::push_constant]]` vs `[[d3d::push_constant]]` or what-have-you.